client/src/js/components/bhDateInterval.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * @name bhDateInterval
+ * @description
+ * The `bhDateInterval` component provide a mean to select dates plage between
+ * two dates. The dates values returned are send to dates models given in
+ * date-from and date-to attributes
+ *
+ * @example
+ * ```html
+ * <bh-date-interval date-from="$MyCtrl.dateFrom" date-to="$MyCtrl.dateTo" >
+ * </bh-date-interval>
+ * ```
+ */
+angular.module('bhima.components').component('bhDateInterval', {
+  templateUrl : '/partials/templates/bhDateInterval.tmpl.html',
+  controller : bhDateInterval,
+  bindings : {
+    dateFrom : '=',
```

Comment:
In general, we should start to avoid using `=` in our components.  However, I do not have a good proposal for how to improve this component, so I'l let it in.